File: OOAD/tictactoe-app/src/components/Cell.java
package components;

import components.enums.Mark;
import components.exception.CellAlreadyMarkedException;

public class Cell {
	private Mark mark;

	public Cell() {
		mark = Mark.EMPTY;
	}

	public Mark getMark() {
		return mark;
	}

	@SuppressWarnings("static-access")
	public void setMark(Mark mark) {
		if (this.mark != this.mark.EMPTY) {
			throw new CellAlreadyMarkedException();
		}
		this.mark = mark;
	}

	@Override
	public String toString() {
		return this.mark.toString();
	}
}

File: C#/MVC/TicTacToe_Layered_With_Unit_Testing-App/TicTacToe_Lib/Components/Cell.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TicTacToe_Lib.Enum;
using TicTacToe_Lib.Exceptions;

namespace TicTacToe_Lib.Components
{
    public class Cell
    {
        private Mark _mark;

        public Cell()
        {
            _mark = Mark.EMPTY;
        }

        public Mark GetMark()
        {
            return _mark;
        }

        public void SetMark(Mark mark)
        {
            if (this._mark != Mark.EMPTY)
            {
                throw new CellAlreadyMarkedException();
            }
            this._mark = mark;
        }


        public override string ToString()
        {
            return this._mark.ToString();
        }
    }
}
